File: app/src/main/java/com/attendance/splash/SplashActivity.java
package com.attendance.splash;

import com.attendance.R;
import com.attendance.courses.CoursesActivity;
import com.attendance.login.LoginActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.WindowManager;
import android.app.Activity;

public class SplashActivity extends Activity implements SplashContract.View {

    private SplashContract.Presenter presenter;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        if (!isTaskRoot()) {
            finish();
            return;
        }
        presenter = new SplashPresenter(this);
        presenter.startTimeTask();
    }

    @Override
    public void setPresenter(SplashContract.Presenter presenter) {
        this.presenter = presenter;
    }

    @Override
    public void startLoginActivity() {
        Intent intent = new Intent();
        intent.setClass(SplashActivity.this,
                LoginActivity.class);
        startActivity(intent);
        finish();
        overridePendingTransition(android.R.anim.fade_in,
                android.R.anim.fade_out);
    }

    @Override
    public void startMainActivity() {
        Intent intent = new Intent();
        intent.setClass(SplashActivity.this,
                CoursesActivity.class);
        startActivity(intent);
        finish();
        overridePendingTransition(android.R.anim.fade_in,
                android.R.anim.fade_out);
    }

}

File: app/src/main/kotlin/com.attendance/splash/SplashActivity.kt
package com.attendance.splash

import com.attendance.R
import com.attendance.splash.SplashContract
import com.attendance.splash.SplashPresenter

import android.content.Intent
import android.os.Bundle
import android.view.WindowManager
import android.app.Activity
import com.attendance.courses.CoursesActivity
import com.attendance.login.LoginActivity

class SplashActivity : Activity(), SplashContract.View {

    private var presenter: SplashContract.Presenter? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN)
        if (!isTaskRoot) {
            finish()
            return
        }
        presenter = SplashPresenter(this)
        presenter!!.startTimeTask()
    }

    override fun setPresenter(presenter: SplashContract.Presenter) {
        this.presenter = presenter
    }

    override fun startLoginActivity() {
        val intent = Intent()
        intent.setClass(this@SplashActivity,
                LoginActivity::class.java)
        startActivity(intent)
        finish()
        overridePendingTransition(android.R.anim.fade_in,
                android.R.anim.fade_out)
    }

    override fun startMainActivity() {
        val intent = Intent()
        intent.setClass(this@SplashActivity,
                CoursesActivity::class.java)
        startActivity(intent)
        finish()
        overridePendingTransition(android.R.anim.fade_in,
                android.R.anim.fade_out)
    }

}